cumulusci/tasks/github/release_report.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+import json
+import pytz
+import re
+import time
+
+from cumulusci.core.utils import process_bool_arg
+from cumulusci.tasks.github.base import BaseGithubTask
+
+
+class ReleaseReport(BaseGithubTask):
+    task_options = {
+        'date_start': {
+            'description': 'Filter out releases created before this date (YYYY-MM-DD)',
+        },
+        'date_end': {
+            'description': 'Filter out releases created after this date (YYYY-MM-DD)',
+        },
+        'include_beta': {
+            'description': 'Include beta releases in report [default=False]',
+        },
+        'print': {
+            'description': 'Print info to screen as JSON [default=False]',
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(ReleaseReport, self)._init_options(kwargs)
+        self.options['date_start'] = self._parse_datetime(
+            self.options['date_start'],
+        ) if 'date_start' in self.options else None
+        self.options['date_end'] = self._parse_datetime(
+            self.options['date_end'],
+        ) if 'date_end' in self.options else None
+        self.options['include_beta'] = process_bool_arg(
+            self.options.get('include_beta', False))
+        self.options['print'] = process_bool_arg(
+            self.options.get('print', False))
+
+    @staticmethod
+    def _parse_datetime(dt_str):
+        t = time.strptime(dt_str, '%Y-%m-%d')
+        return datetime(t[0], t[1], t[2], t[3], t[4], t[5], t[6], pytz.UTC)
+
+    def _run_task(self):
+        releases = []
+        last_time = None
+        repo = self.get_repo()
+        prog = re.compile(
+            r'^((?P<sandbox>{})|(?P<production>{}))\s*(?P<date>\d\d\d\d-\d\d-\d\d)'.format(
+                self.project_config.project__git__push_prefix_sandbox,
+                self.project_config.project__git__push_prefix_production,
+            ))
+        for release in repo.iter_releases():
+            if release.prerelease and not self.options['include_beta']:
+                continue
+            if self.options['date_start'] and release.created_at < self.options[
+                    'date_start']:
+                continue
+            if self.options[
+                    'date_end'] and release.created_at > self.options['date_end']:
```

Comment:
should these ranges be inclusive? (i.e. >= ) I think so...? at least one side should be right?

Reply:
This is basically "use release if start_date <= release_date <= end_date". It's just flipped around to the negative form because the `continue` statements skip the releases.

Reply:
got it